src/petals/server/server.py

```diff
@@ -250,7 +250,12 @@ def _choose_num_blocks(self) -> int:
         # Estimate of GPU memory used in rpc_backward (2 GiB for BLOOM, proportional for other models)
         autograd_memory = 2 * gib * num_devices / 14336 * self.block_config.hidden_size
 
-        num_blocks = math.floor((total_memory - autograd_memory) / (block_size + self._cache_bytes_per_block))
+        adapter_memory_per_block = estimate_adapter_memory_per_block(
+            self.block_config, self.torch_dtype, self.adapters, self.cache_dir
+        )
```

Comment:
```suggestion
        if adapters:
            # Delay import of petals.utils.peft to avoid unnecessary import of bitsandbytes
            from petals.utils.peft import estimate_adapter_memory_per_block
            
            adapter_memory_per_block = estimate_adapter_memory_per_block(
                self.block_config, self.torch_dtype, self.adapters, self.cache_dir
            )
```